run_galaxy_workflow.py

```diff
@@ -349,8 +354,11 @@ def main():
 
         # upload dataset to history
         logging.info('Uploading dataset to history ...')
-        datamap = load_input_files(gi, inputs=inputs_data,
+        if num_inputs > 0:
+            datamap = load_input_files(gi, inputs=inputs_data,
                                    workflow=show_wf, history=history)
+        else:
+            datamap = {}
```

Comment:
While uploading datasets to history we are not setting parameters explicitly as I see in the else part parameters settings happening.

Reply:
This is in the case when there are no inputs files, hence nothing to upload, and so the datamap should be empty. Datamap (for inputs) and parameters are independent of course.